### PR TITLE
Make `Prop3DMixin.bounds` abstract

### PR DESCRIPTION
**Pull request by user27182**
### Overview

#6412 added `bounds` to `_Prop3DMixin` with a concrete implementation, but this property should be abstract since the bounds rely on a transformation applied by the abstract method `_post_set_update`.

The current implementation for `bounds` was actually  based on two separate concrete implementations for the `_XYZAssembly` and `Label` classes, so this PR splits and moves the implementation to those classes. To avoid the need for subclasses to re-write or re-wrap the docs for `bounds` , a new `_get_bounds` method is used for the abstraction and called internally instead of making the `bounds`  property itself abstract.

### CODE REVIEW

File: tests/plotting/test_text.py [hunk 1]
```diff
@@ -72,6 +72,12 @@ def test_label_prop3d():
     assert label.center == position
     assert label.length == 0.0
 
+    # Test correct bounds with more complex transformations
+    label.relative_position = (100, 0, 0)
+    label.orientation = (0, 0, 90)
+    bounds = (1.0, 1.0, 102.0, 102.0, 3.0, 3.0)
+    assert np.allclose(label.bounds, bounds)
```
Comment (tkoyama010) [inline tests/plotting/test_text.py:79]: Please add a comment explaining why the `bounds` have this value.

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: tests/plotting/test_text.py (hunk 1 above)
Comment (user27182) [inline tests/plotting/test_text.py:79]: Is this better?

```suggestion
    # Test correct bounds with more complex transformations
    # Add offset along x-axis
    offset = 100
    label.relative_position = (offset, 0, 0)
    # Rotate about z-axis
    label.orientation = (0, 0, 90)
    # Expect offset to be applied along y-axis (due to the rotation) 
    bounds = (position[0], position[0], position[1] + offset, position[1] + offset, position[2], position[2])
    assert np.allclose(label.bounds, bounds)
```

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**